spec/classes/repo_spec.rb

```diff
@@ -27,17 +27,17 @@
           it { is_expected.to contain_apt__source('zabbix').with_location('https://example.com/foo') }
         end
 
-        case facts[:os]['release']['major']
-        when '6'
-          context 'on Debian 6 and Zabbix 2.0' do
+        case facts[:os]['release']['major'].to_i
+        when 10
```

Comment:
I would avoid `.to_i`. It's always a string and your code doesn't become easier with integers
```suggestion
        case facts[:os]['release']['major']
        when '10'
```

You also introduce a subtle bug on line 43 which now no longer matches.